plugin/src/main/java/org/opennms/plugins/cloud/tsaas/TsaasStorage.java

```diff
@@ -168,14 +169,25 @@ private static Timestamp toTimestamp(Instant instant) {
 
     @Override
     public void store(List<Sample> samples) throws StorageException {
-        List<Tsaas.Sample> mappedSamples = samples.stream()
+
+        // convert given samples to grpc
+        samples.stream()
                 .map(TsaasStorage::toSample)
-                .collect(Collectors.toList());
-        Tsaas.Samples samplesMessage = Tsaas.Samples.newBuilder()
-                .addAllSamples(mappedSamples)
-                .build();
-        LOG.trace("Storing the following samples: {}", samplesMessage);
-        clientStub.store(samplesMessage);
+                .forEach(this.queue::add);
+
+        // send batched messages while the queue is fuller than batch size
+        while(this.queue.size() >= this.config.getBatchSize()) {
```

Comment:
We should also consider adding a timer here that will flush this periodically: imagine sending 1001 samples, the 1000 will be flushed, and then the 1 will sit there... until 999 more show up.